ner_v1/detectors/textual/name/name_detection.py

```diff
@@ -227,6 +288,176 @@ def context_check_botmessage(botmessage):
+            2. original_text (list): representing the original text detected
+
+
+        Example:
+            text = u'प्रतिक श्रीदत्त जयराओ'
+            get_hindi_names_without_regex(text=text)
+
+            >> [{first_name: u"प्रतिक", middle_name: u"श्रीदत्त", last_name: u"जयराओ"}], [ u'प्रतिक श्रीदत्त जयराओ']
+
+        """
+        original_text_list = self.get_devnagri_text(text=text)
+        replaced_text = self.replace_detected_text((original_text_list, original_text_list))
+        return self.detect_person_name_entity(replaced_text=replaced_text)
+
+    def get_devnagri_text(self, text):
+        """
+        This method is used to detect devnagri text from the text and additionally check is
+        length is less than 5
+        Args:
+            text (str): the string from which devnagri text has to be detected
+        Returns:
+            split_list (list): list of devanagri text split on basis of whitespaces
+
+        Examples:
+            text = u'प्रतिक श्रीदत्त जयराओ'
+            get_devnagri_text(text=text)
+            >>[u'प्रतिक', u'श्रीदत्त', u'जयराओ']
+        """
+        text = self.replace_stopwords_hindi(text)
+        regex = re.compile(ur"[^\u0900-\u097F\s]+", re.U)
+        text = regex.sub(string=text, repl='')
+        split_text = text.split()
+        if len(split_text) <= 4:
+            return split_text
+
+    def get_hindi_text_from_regex(self, text):
+        """
+        This method is used to detect hindi names using regexes from the given text
+        Args:
+            text (str): text from which hindi names which follow the regex pattern have to be extracted
+
+        Returns:
+            pattern_match (list): list consisting of detected words
+
+        Examples:
+            text = u'मेरा नाम प्रतिक श्रीदत्त जयराओ है'
+            get_hindi_text_from_regex(text=text)
+            >>[u'प्रतिक श्रीदत्त जयराओ']
+
+        """
+        regex_list = [ur"(?:नाम|मैं|हम)\s*([\u0900-\u097F\s]+)",
+                      ur"(?:मुझे|हमें|मुझको|हमको)\s*([\u0900-\u097F\s]+)(?:कहते|बुलाते|बुलाओ)",
+                      ur"\s*([\u0900-\u097F\s]+)(?:मुझे|मैं)(?:कहते|बुलाते|बुलाओ)?"
+                      ]
+
+        for regex in regex_list:
+            regex_ = re.compile(regex, re.U)
+            pattern_match = regex_.findall(text)
+            pattern_match = [self.replace_stopwords_hindi(x) for x in pattern_match if x]
+            if pattern_match:
+                if None != pattern_match[0]:
+                    return pattern_match
+
+    def replace_stopwords_hindi(self, text):
+        """
+        This method is used to replace hindi stop words from the text
+        Args:
+            text (str): The text from which hindi stop words have to be removed
+
+        Returns:
+            clean_text (str): text from which hindi stop words have been removed
+        """
+        split_list = text.split(" ")
+        split_list = [word for word in split_list if word not in HINDI_STOPWORDS]
```

Comment:
where is else cond?
you should return None or empty list

Reply:
Done